test/v4/orderbookReserve.js

```diff
@@ -3268,501 +3187,491 @@ contract('OrderbookReserve', async (accounts) => {
-            let tokenWeiDepositAmount = new BigNumber(0 * 10 ** 18);
-            let kncTweiDepositAmount = 600 * 10 ** 18;
-            let ethWeiDepositAmount = (new BigNumber(10 * 10 ** 18)).add(30000);
+            let tokenWeiDepositAmount = PRECISION.mul(new BN(0));
+            let kncTweiDepositAmount = PRECISION.mul(new BN(600));
+            let ethWeiDepositAmount = PRECISION.mul(new BN(10)).add(new BN(30000));
             await makerDeposit(maker1, ethWeiDepositAmount, tokenWeiDepositAmount, kncTweiDepositAmount);
 
-            let orderTwei = new BigNumber(9 * 10 ** 18);
-            let orderWei = new BigNumber(2 * 10 ** 18);
+            let orderTwei = PRECISION.mul(new BN(9));
+            let orderWei = PRECISION.mul(new BN(2));
 
             await reserve.submitEthToTokenOrder(orderWei, orderTwei, {from: maker1});
-            let rc = await reserve.submitEthToTokenOrder(orderWei, orderTwei.add(100), {from: maker1});
-            let id2ndOrder = rc.logs[0].args.orderId.valueOf();
-            rc = await reserve.submitEthToTokenOrder(orderWei, orderTwei.add(200), {from: maker1});
-            let id3rdOrder = rc.logs[0].args.orderId.valueOf();
-            await reserve.submitEthToTokenOrder(orderWei, orderTwei.add(300), {from: maker1});
-            await reserve.submitEthToTokenOrder(orderWei, orderTwei.add(400), {from: maker1});
+            let rc = await reserve.submitEthToTokenOrder(orderWei, orderTwei.add(new BN(100)), {from: maker1});
+            let id2ndOrder = rc.logs[0].args.orderId;
+            rc = await reserve.submitEthToTokenOrder(orderWei, orderTwei.add(new BN(200)), {from: maker1});
+            let id3rdOrder = rc.logs[0].args.orderId;
+            await reserve.submitEthToTokenOrder(orderWei, orderTwei.add(new BN(300)), {from: maker1});
+            await reserve.submitEthToTokenOrder(orderWei, orderTwei.add(new BN(400)), {from: maker1});
 
             await reserve.cancelEthToTokenOrder(id2ndOrder, {from: maker1});
             await reserve.cancelEthToTokenOrder(id3rdOrder, {from: maker1});
 
-            await reserve.submitEthToTokenOrderWHint(orderWei, orderTwei.add(150), id3rdOrder, {from: maker1});
+            await reserve.submitEthToTokenOrderWHint(orderWei, orderTwei.add(new BN(150)), id3rdOrder, {from: maker1});
 
             let orderList = await reserve.getEthToTokenOrderList();
 
-            assert.equal(orderList.length, 4);
+            Helper.assertEqual(orderList.length, 4);
         })
     });
 
     describe("knc stakes and burn", function() {
         it("maker add sell token order. see funds & knc stakes updated.", async () => {
-            let kncTweiDepositAmount = new BigNumber(600 * 10 ** 18);
-            let tokenWeiDepositAmount = new BigNumber(11.1 * 10 ** 18);
+            let kncTweiDepositAmount = PRECISION.mul(new BN(600));
+            let tokenWeiDepositAmount = new BN(String(PRECISION * 11.1));
```

Comment:
does string(PRECISION * 11.1) give same result as 11.1 * 10 ** 18 ??

Reply:
Oh
I see it is
nice solution.

Reply:
Yeah, this is the simplest solution to make it work. BN doesn't understand decimals. So if you do `new BN(11.1)`, it will round it off to 11.